zipline/src/commonMain/kotlin/app/cash/zipline/internal/bridge/calls.kt

```diff
@@ -296,3 +297,78 @@ internal class ResultSerializer<T>(
     }
   }
 }
+
+/**
+ * Immediate result from invoking a suspending function, that may include either a value (if the
+ * call never suspended) or a cancel callback (if it did suspend).
+ */
+internal class SuspendingResult<T>(
```

Comment:
Considering for follow-up . . . 

We have a bunch of result types:
 - kotlin.Result - success or failure
 - app.cash.zipline.CallResult - success or failure, encoded JSON, service names
 - app.cash.zipline.internal.bridge.SuspendingResult - success or failure or callback
 - app.cash.zipline.internal.bridge.EncodedSuspendingResult - success or failure or callback, encoded JSON, service names

We also have serializers for kotlin.Result and SuspendingResult.

Some options to reduce how much code we need:
 * Combine the two serializers to save a bit of code. Rename SuspendingResult to InternalResult for symmetry with InternalCall.
 * Change EncodedSuspendingResult to have a `kotlin.Result` as a field